amethyst_ui/src/event.rs

```diff
@@ -133,3 +138,31 @@ where
         self.was_down = down;
     }
 }
+
+fn targeted(
+    pos: (f32, f32),
+    transforms: JoinIter<(
```

Comment:
If you don't want to use the internal specs types here, you could make it something like:

```
fn targeted<'a, I>(pos: (f32, f32), transforms: I, react: &ReadStorage<MouseReactive>) 
where 
    I: Iterator<Item=(Entity, &'a UiTransform)> + 'a,
{
    ..
}
```

Reply:
That way you don't need `MaskedStorage` etc.

Reply:
Fixed to use this in my rebase.